run_page/utils.py

```diff
@@ -117,3 +117,23 @@ def upload_file_to_strava(client, file_name, data_type, force_to_run=True):
         print(
             f"Uploading {data_type} file: {file_name} to strava, upload_id: {r.upload_id}."
         )
+
+
+def upload_file_to_strava_with_type(client, file_name, data_type, type):
+    with open(file_name, "rb") as f:
+        try:
+            r = client.upload_activity(
+                activity_file=f, data_type=data_type, activity_type=type
```

Comment:
I wonder how Strava determines the type if you just upload the file but don't specify the activity_type?
For example, if you upload a tcx with the type of `hiking`, what type would it be in Strava?

Reply:
1. got the description in Stravalib API here:
`
       :param activity_type: (optional) case-insensitive type of activity.
                            possible values: ride, run, swim, workout, hike, walk,
                            nordicski, alpineski, backcountryski, iceskate, inlineskate,
                            kitesurf, rollerski, windsurf, workout, snowboard, snowshoe
                            Type detected from file overrides, uses athlete's default type if not specified
      :type activity_type: str
`

2. I have performed a test on my pc, and the result is just like uploading failed without strong hint:
`
to load all tcx files maybe take some time
Access ok
Start syncing
.
Process finished with exit code 0
`

Reply:
I've checked the documentation for the upload API, which states that it detects the `sport type` from the file. refer: https://developers.strava.com/docs/uploads/
![image](https://github.com/yihong0618/running_page/assets/6956444/3ec0e7d6-af94-4d50-9c76-b817b0fc5096)




> 2. I have performed a test on my pc, and the result is just like uploading failed without strong hint:
> 
> ```
> to load all tcx files maybe take some time
> Access ok
> Start syncing
> .
> Process finished with exit code 0
> ```
A dot (`.`) in the console indicates an updated activity, and a plus (`+`) indicates a created activity. I'm guessing the file you uploaded was already synced to Strava. 
Can you help me test the following steps: 
1. Export a TCX of an activity in a others type, such as hiking or cycling. 
2. Delete the activity in Strava (you can re-upload it with the exported TCX file). 
4. Upload it using `utils.upload_file_to_strava` with `force_to_run = False` 
5. Check the activity type uploaded to Strava. if the type is correct then I believe we can use this function instead of adding `upload_file_to_strava_with_type`.

Reply:
I have tested the TCX format of 'Running' and 'Biking'. The result is just as you said, and the type which shows at Strava is correct.

So, It seems that this PR is no longer needed?
@ben-29